### ISSUE TITLE
binarytrees solution is wrong

### ISSUE THREAD

**Issue by vanyakosmos** [golang/tour#874]
https://github.com/golang/tour/blob/3765179383a5cd54eed800c62cc4b9de24fbce0b/solutions/binarytrees.go#L43-L50

If last elements are **not equal** but both channels are closed then `Same` function will return `true`, which is wrong. `if v1 != v2` and `if !ok1 || !ok2` should be switched.

**Comment by ALTree** [comment]
I don't think you can compare v1 and v2 is the channels are closed, they'll hold no values.
